pkg/machine/e2e/cp_test.go

```diff
@@ -0,0 +1,214 @@
+package e2e_test
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gexec"
+)
+
+var _ = Describe("podman machine cp", func() {
+	It("host file to guest", func() {
+		// initialize a machine and start it
+		name := randomString()
+		initMachine := initMachine{}
+		session, err := mb.setName(name).setCmd(initMachine.withImage(mb.imagePath).withNow()).run()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(session).To(Exit(0))
+
+		// create a file on the host in the temp directory
+		path := filepath.Join(GinkgoT().TempDir(), "tmp.txt")
+		tempFile, err := os.Create(path)
+		Expect(err).ToNot(HaveOccurred())
+		defer tempFile.Close()
+
+		// copy the file into the guest
+		cp := cpMachine{}
+		session, err = mb.setCmd(cp.withQuiet().withSrc(path).withDest(name + ":~/tmp.txt")).run()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(session).To(Exit(0))
+
+		// verify the guest has the file
+		sshMachine := sshMachine{}
+		session, err = mb.setName(name).setCmd(sshMachine.withSSHCommand([]string{"ls"})).run()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(session).To(Exit(0))
+		Expect(session.outputToString()).To(Equal("tmp.txt"))
+
+		// try to copy the file to a location in the guest where permission will get denied
+		session, err = mb.setCmd(cp.withQuiet().withSrc(path).withDest(name + ":/etc/tmp.txt")).run()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(session).To(Exit(125))
+		Expect(session.errorToString()).To(ContainSubstring("scp: dest open \"/etc/tmp.txt\": Permission denied"))
+	})
+
+	It("host directory to guest", func() {
+		// initialize a machine and start it
+		name := randomString()
+		initMachine := initMachine{}
+		session, err := mb.setName(name).setCmd(initMachine.withImage(mb.imagePath).withNow()).run()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(session).To(Exit(0))
+
+		// create a directory in the temp dir
+		dirPath := filepath.Join(GinkgoT().TempDir(), "foo")
+		err = os.MkdirAll(dirPath, 0755)
+		Expect(err).ToNot(HaveOccurred())
+		defer os.RemoveAll(dirPath)
```

Comment:
as long as you use `GinkgoT().TempDir(` there is no need to defer remove, the whole point of `TempDir()` is that we do not have to clutter our tests with the defers